app/models/merchant.rb

```diff
@@ -50,4 +50,24 @@ def top_sales_day
       .created_at
       .strftime("%-m/%-e/%Y")
   end
-end
\ No newline at end of file
+
+  def top_5_popular_items
+    items.select("items.*, sum(invoice_items.quantity * invoice_items.unit_price) as total_item_revenue")
+    .joins(:merchant, invoices: :transactions)
+    .where("transactions.result = ?", 0)
+    .group("items.id")
+    .order("total_item_revenue DESC")
+    .limit(5)
+  end
```

Comment:
A fun fact I learned from top 5 merchants is apparently, if the selected sum(# * price)'s alias is total_revenue, the format_total_revenue method in the application_record.rb file can be called on it.